traits/tests/test_trait_list_object.py

```diff
@@ -364,3 +865,207 @@ class Foo:
+            foo.at_most_five.append(6)
+        self.assertEqual(foo.at_most_five, [1, 2, 3, 4, 5])
+
+    def test_clear(self):
+        foo = HasLengthConstrainedLists(at_most_five=[1, 2, 3, 4])
+        foo.at_most_five.clear()
+        self.assertEqual(foo.at_most_five, [])
+
+    def test_clear_too_small(self):
+        foo = HasLengthConstrainedLists(at_least_two=[1, 2, 3, 4])
+        with self.assertRaises(TraitError):
+            foo.at_least_two.clear()
+        self.assertEqual(foo.at_least_two, [1, 2, 3, 4])
+
+    def test_extend(self):
+        foo = HasLengthConstrainedLists(at_least_two=[1, 2, 3, 4])
+        foo.at_least_two.extend([10, 11])
+        self.assertEqual(foo.at_least_two, [1, 2, 3, 4, 10, 11])
+
+    def test_extend_too_large(self):
+        foo = HasLengthConstrainedLists(at_most_five=[1, 2, 3, 4])
+        with self.assertRaises(TraitError):
+            foo.at_most_five.extend([10, 11, 12])
+        self.assertEqual(foo.at_most_five, [1, 2, 3, 4])
+
+    def test_extend_from_iterable(self):
+        foo = HasLengthConstrainedLists(at_most_five=[1, 2])
+        foo.at_most_five.extend(squares(3))
+        self.assertEqual(foo.at_most_five, [1, 2, 0, 1, 4])
+
+    def test_insert(self):
+        foo = HasLengthConstrainedLists(at_least_two=[1, 2, 3, 4])
+        foo.at_least_two.insert(3, 16)
+        self.assertEqual(foo.at_least_two, [1, 2, 3, 16, 4])
+
+    def test_insert_too_large(self):
+        foo = HasLengthConstrainedLists(at_most_five=[1, 2, 3, 4, 5])
+        with self.assertRaises(TraitError):
+            foo.at_most_five.insert(3, 16)
+        with self.assertRaises(TraitError):
+            foo.at_most_five.insert(-10, 16)
+        with self.assertRaises(TraitError):
+            foo.at_most_five.insert(10, 16)
+        self.assertEqual(foo.at_most_five, [1, 2, 3, 4, 5])
+
+    def test_pop(self):
+        foo = HasLengthConstrainedLists(at_least_two=[1, 2, 6])
+        foo.at_least_two.pop()
+        self.assertEqual(foo.at_least_two, [1, 2])
+
+    def test_pop_too_small(self):
+        foo = HasLengthConstrainedLists(at_least_two=[1, 2])
+        with self.assertRaises(TraitError):
+            foo.at_least_two.pop()
+        with self.assertRaises(TraitError):
+            foo.at_least_two.pop(0)
+        # TraitError takes precedence over the IndexError for a bad index.
+        with self.assertRaises(TraitError):
+            foo.at_least_two.pop(10)
+        self.assertEqual(foo.at_least_two, [1, 2])
+
+    def test_remove(self):
+        foo = HasLengthConstrainedLists(at_least_two=[1, 2, 6, 4])
+        foo.at_least_two.remove(2)
+        self.assertEqual(foo.at_least_two, [1, 6, 4])
+
+    def test_remove_too_small(self):
+        foo = HasLengthConstrainedLists(at_least_two=[1, 2])
+        with self.assertRaises(TraitError):
+            foo.at_least_two.remove(1)
+        with self.assertRaises(TraitError):
+            foo.at_least_two.pop(2.0)
```

Comment:
Is this supposed to be `foo.at_least_two.remove(2.0)`?

Reply:
D'oh! Fixed.